src/impl/MilvusClientImpl.cpp

```diff
@@ -472,7 +472,17 @@ MilvusClientImpl::Delete(const std::string& collection_name, const std::string&
 
 Status
 MilvusClientImpl::Search(const SearchArguments& arguments, SearchResults& results) {
-    auto pre = [&arguments]() {
+    std::string anns_field;
+    auto validate = [this, &arguments, &anns_field]() {
+        CollectionDesc collection_desc;
+        auto status = DescribeCollection(arguments.CollectionName(), collection_desc);
+        if (status.IsOk()) {
+            anns_field = collection_desc.Schema().AnnsFieldName();
```

Comment:
I suggest add a method "SetAnnsFieldName()" to the SearchArguments class, or change the AddTargetVectors() to be:
`Status AddTargetVector(const std::string& field_name, const FloatVecFieldData::ElementT& vector)`

Then create FieldData with the name, if the FieldData already exists, just return an error. 
Now we only allow search one vector field, in someday maybe we can allow two vectors fields to search together with  different weight.

Reply:
Seems `Status AddTargetVector(const std::string& field_name, const FloatVecFieldData::ElementT& vector)` is more reasonable right now.